packages/manager/modules/iam/src/components/actionSelect/actionSelect.controller.js

```diff
@@ -0,0 +1,303 @@
+import { cloneDeep } from 'lodash-es';
+import ActionTrees from './ActionTrees.class';
+
+export default class ActionSelectController {
+  /* @ngInject */
+  constructor($attrs, $scope, $timeout, $translate, ReferenceService) {
+    this.$attrs = $attrs;
+    this.$scope = $scope;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.ReferenceService = ReferenceService;
+
+    /**
+     * List of actions
+     * @type {Object[]}
+     */
+    this.actions = [];
+
+    /**
+     * Group each action into categories and each category into resource types
+     * So it can be displayed as collapsible components
+     * @type {ActionTrees=}
+     */
+    this.actionTrees = undefined;
```

Comment:
`undefined` is used so the property can be destructured with a default value (`null` is not considered as optional)